apps/graphql/src/apps/itinerary/dataloaders/Itinerary.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+import stringify from 'json-stable-stringify';
+import qs from 'querystring';
+import { OptimisticDataloader } from '@kiwicom/graphql-utils';
+
+import fetch from '../../../services/fetch/Fetch';
+import type {
+  ItineraryCheckParameters,
+  ItineraryApiResponseType,
+  Itinerary,
+} from '../Itinerary';
+
+export const parseParameters = (input: ItineraryCheckParameters) => {
+  const passengersTotal =
+    (input.passengers.adults ?? 0) +
+    (input.passengers.children ?? 0) +
+    (input.passengers.infants ?? 0);
+
+  const params = {
+    booking_token: input.bookingToken,
+    bnum: input.bags,
+    pnum: passengersTotal,
+  };
+  return params;
+};
+
+const fetchItinerary = async (
+  parameters: $ReadOnlyArray<ItineraryCheckParameters>,
+) => {
+  const results: $ReadOnlyArray<ItineraryApiResponseType> = await Promise.all(
+    parameters.map(params => {
+      return fetch(
+        `/v2/booking/check_flights?${qs.stringify(parseParameters(params))}`,
+      );
+    }),
+  );
+  return results.map(res => {
+    return sanitizeItinerary(res);
+  });
+};
+
+const sanitizeItinerary = (response: ItineraryApiResponseType): Itinerary => {
+  // @TODO - map remaining (null) data
+  return {
+    id: mapItineraryId(response.flights),
+    type: null,
+    bookingToken: response.booking_token,
+    isValid: !response.flights_invalid,
+    isChecked: response.flights_checked,
+    startTime: null,
+    endTime: null,
+    origin: null,
+    destination: null,
+    sectors: null,
+    price: {
+      currency: 'EUR',
```

Comment:
pls add TODO comment also here

Reply:
Right, updated. The price part on this endpoint is slightly confusing, because it always returns total price just in EUR. I have few other notes from fiddling with API, so I'll add them to our Tequila related issue after I clean them a bit.

Reply:
thanks